src/main/kotlin/org/rust/lang/utils/RsDiagnostic.kt

```diff
@@ -35,12 +37,19 @@ sealed class RsDiagnostic(
         private val expectedTy: Ty,
         private val actualTy: Ty
     ) : RsDiagnostic(element, experimental = true) {
-        override fun prepare() = PreparedAnnotation(
-            ERROR,
-            E0308,
-            "mismatched types",
-            expectedFound(expectedTy, actualTy)
-        )
+        override fun prepare(): PreparedAnnotation {
+            val fixes = mutableListOf<LocalQuickFix>()
+            if (expectedTy is TyNumeric && isActualTyNumeric()) fixes.add(AddAsTyFix(element, expectedTy))
+            return PreparedAnnotation(
+                ERROR,
+                E0308,
+                "mismatched types",
+                expectedFound(expectedTy, actualTy),
+                fixes = fixes
```

Comment:
Try to use our `buildList` function here, it may look much cleaner.

Reply:
done